### DISCUSSION TITLE
How to display a derivative inside a recursive function

### DISCUSSION THREAD

**Discussion by koblez** [wxMaxima-developers/wxmaxima#1719] [Q&A]
Hello, I wrote the following function that calculates the 5th derivative of a function f(x,y,z) but the print() inside the functions doesn't display the result, instead only d/dx(d/dy(d/dy)) ... etc is displayed.
The function:
```
f(x,y,z):=x^2*y^2-x*y*z^2-3*x^2*z+5*z^4+%e^y;

fdiff(expr, i):=block(
    [k],
    k:0,
    if i=5 then print(expr)
    else
        for k step 1 thru 3 do
            if k=1 then
                fdiff(diff(expr(x,y,z),x,1),i+1)
            else if k=2 then
                fdiff(diff(expr(x,y,z),y,1),i+1)
            else if k=3 then
                fdiff(diff(expr(x,y,z),z,1),i+1)
);
fdiff(f(x,y,z),0);
```

How do I display the results correctly ?

**Comment by daute** [comment]
[Remark: This question seems to be a Maxima issue, not related to wxMaxima, the graphical frontend.]

What do you want? Just the fifth derivative? The `diff()` function supports a parameter to specify the n'th derivative:

    diff (<expr>, <x>, <n>) returns the <n>'th derivative of <expr> with respect to <x>.

So if you want the fifth derivative with respect to z (the result will be zero!), just do

    diff(f(x,y,z),z,5);

If you want to print the first, second, third, fourth and fifth derivative, use

    for k from 1 thru 5 do print(diff(f(x,y,z),z,k));
    
Is that the result you wanted?
Best regards, Wolfgang

**Reply by koblez** [reply]
Sorry for the late response 😅.
And as you wrote, this questions isn't really wxMaxima related, but after some hours of searching, this discussions board was the easiest for me to access, because other maxima websites need manual admin approval for new accounts and I'm still waiting for that to happen...
I wanted the fifth derivative of every possible permutation of xyz so something like this:
f_xxxxx(x,y,z) (would be, as you wrote, `diff(f(x,y,z),x,5)`)
f_xxxxy(x,y,z)
f_xxxxz(xy,z)
f_xxxyx(x,y,z)
...
f_xzxyy(x,y,z)
...
f_zzzzz(x,y,z)
That would make 243 derivatives.
I know that e.g. f_xxxxz(x,y,z) == f_xxzxx(x,y,z), I just wanted to experiment with the functionality of (wx)Maxima.


Best regards!
